ArduCopter/mode_auto.cpp

```diff
@@ -1970,4 +1970,22 @@ bool ModeAuto::verify_nav_delay(const AP_Mission::Mission_Command& cmd)
     return false;
 }
 
+/********************************************************************************/
+// Pause and continue auto mode
+/********************************************************************************/
+
+// pause_mission - Prevent aircraft from progressing along the track
+bool ModeAuto::pause()
+{
+    wp_nav->set_pause();
```

Comment:
I think we should return false if we're not in the WP sub mode and !reached_destination.  Same goes for "resume".  This allows us to only handle SCurves and splines and forget about all the other sub modes.

Reply:
Done.